Library/Homebrew/rubocops/cask/mixin/cask_help.rb

```diff
@@ -13,14 +13,20 @@ module CaskHelp
         sig { abstract.params(cask_block: RuboCop::Cask::AST::CaskBlock).void }
         def on_cask(cask_block); end
 
+        def on_cask_stanza_block(cask_stanza_block); end
+
         def on_block(block_node)
           super if defined? super
-          return unless respond_to?(:on_cask)
-          return unless block_node.cask_block?
 
-          comments = processed_source.comments
-          cask_block = RuboCop::Cask::AST::CaskBlock.new(block_node, comments)
-          on_cask(cask_block)
+          if respond_to?(:on_cask_stanza_block) && (block_node.cask_block? || block_node.on_system_block?)
+            on_cask_stanza_block(block_node)
```

Comment:
Here, `block_node` can now be a `cask` or a `on_*` block, i.e. they are treated equally without any sort of nested loop.